go/apps/mixnet/mixnet_test.go

```diff
@@ -298,43 +288,48 @@ func TestMaliciousProxyRouterRelay(t *testing.T) {
 		t.Fatal(err)
 	}
 	defer router.Close()
-	ch := make(chan testResult)
-
 	cell := make([]byte, CellBytes)
-	buff := make([]byte, CellBytes*10)
+	ch := make(chan testResult)
 
-	// Unrecognized cell type.
-	cell[0] = dirCell ^ msgCell ^ relayCell
-	go runRouterHandleProxy(router, 1, ch)
+	go runRouterHandleProxy(router, 4, ch)
 	c, err := proxy.DialRouter(network, routerAddr)
 	if err != nil {
 		t.Error(err)
 	}
+
+	// Unrecognized cell type.
+	cell[0] = 0xff
 	if _, err = c.Write(cell); err != nil {
 		t.Error(err)
 	}
-	_, err = proxy.ReceiveMessage(c, buff)
+	_, err = proxy.ReceiveMessage(c)
 	if err == nil || err.Error() != "router error: "+errBadCellType.Error() {
-		t.Error("Bad cell, got incorrect error: ", err)
+		t.Error("bad cell, got incorrect error:", err)
 	}
-	c.Close()
-	<-ch
 
-	// Message too long.
+	// Message too long
 	cell[0] = msgCell
 	binary.PutUvarint(cell[1:], uint64(MaxMsgBytes+1))
-	go runRouterHandleProxy(router, 1, ch)
-	c, err = proxy.DialRouter(network, routerAddr)
-	if err != nil {
+	if _, err := c.Write(cell); err != nil {
 		t.Error(err)
 	}
-	if _, err = c.Write(cell); err != nil {
+	_, err = proxy.ReceiveMessage(c)
+	if err == nil || err.Error() != "router error: "+errMsgLength.Error() {
+		t.Error("message too long, got incorrect error:", err)
+	}
+
+	// Bogus destination
+	if err = proxy.CreateCircuit(c, []string{"localhost:9999"}); err != nil {
```

Comment:
A minor style note for future reference: it's more idiomatic to make this err := proxy.CreateCircuit. This also helps if the code changes around it and this is the first instance of err.